Automattic/wp-calypso
# Add notification unseen count to redux store

This PR adds the notification unseen count to the redux state. 

### Motivation & Context
In WP-Desktop we'd like to subscribe to changes in the redux state for e.g. displaying/updating the dock badge and bouncing the dock icon (+ hundreds of other 😎things). Therefore, being able to subscribe to changes gives us tons of new opportunities for interacting with the desktop app.

The way it's currently shared with wp-desktop is quite faulty as parts of the communication with the app has been deprecated a while ago. (`message` event) 

@dmsnell has been so kind to walk be through the migration process.

### How to test
- Start the dev server and open calypso
- Create a new notification
- In console enter `actionLog.filter(/NOTIFICATIONS/i);`
- You should see something like: 
   ```js
   [{type: "NOTIFICATIONS_UNSEEN_COUNT_SET", unseenCount: 1, meta: {…}}]
   ```

### Gotchas/Todos
- The notification count is only updated when e.g. a new notification is created. We might need to update this during the initial load as well.
- We need to expose the redux state to files like `client/lib/desktop.js` to be able to subscribe to changes. /cc @jsnajdr, @gwwar
